### DISCUSSION TITLE
How to chager url and update image in swiftUI

### DISCUSSION THREAD

**Discussion by hoangnam714** [onevcat/Kingfisher#1979] [Q&A] [answered]
<img width="927" alt="Screen Shot 2022-08-18 at 08 52 56" src="https://user-images.githubusercontent.com/51524705/185275182-09192f72-1e2b-4b09-bf76-3d503083b583.png">
I have an image view that needs to change the image when it is changed, I am having trouble reloading the KFImage when loading with new urls, can someone help me

**Comment by onevcat** [comment, marked as answer]
Here is a sample running well when changing the URL (or URL string):

```swift
struct SingleViewDemo : View {
    @State var string = "kingfisher-1.jpg"
    var body: some View {
        KFImage(
            URL(string: "https://raw.githubusercontent.com/onevcat/Kingfisher-TestImages/master/DemoAppImage/Loading/\(string)")
        )
        Button(action: {
            string = "kingfisher-2.jpg"
        }) { Text("Next Image") }
    }
}
```

![](https://user-images.githubusercontent.com/1019875/185316862-c766435a-8209-44dd-b111-fad751c0a317.mov)

Any luck to try this code and see if that works? If so, maybe you are having a wrong `url` in your `Button` action?

**Comment by hoangnam714** [comment]
Thank @onevcat, Thanks for your reply, have a nice day.
